src/uct/ib/base/ib_device.c

```diff
@@ -399,3 +400,56 @@ ucs_status_t uct_ib_device_query_tl_resources(uct_ib_device_t *dev,
 err:
     return status;
 }
+
+ucs_status_t uct_ib_device_find_port(uct_ib_device_t *dev,
+                                     const char *resource_dev_name,
+                                     uint8_t *p_port_num)
+{
+    const char *ibdev_name;
+    unsigned port_num;
+    size_t devname_len;
+    char *p;
+
+    p = strrchr(resource_dev_name, ':');
+    if (p == NULL) {
+        goto err; /* Wrong device name format */
+    }
+    devname_len = p - resource_dev_name;
+
+    ibdev_name = uct_ib_device_name(dev);
+    if ((strlen(ibdev_name) != devname_len) ||
+        strncmp(ibdev_name, resource_dev_name, devname_len))
+    {
+        goto err; /* Device name is wrong */
+    }
+
+    port_num = strtod(p + 1, &p);
+    if (*p != '\0') {
+        goto err; /* Failed to parse port number */
+    }
+    if ((port_num < dev->first_port) || (port_num >= dev->first_port + dev->num_ports)) {
+        goto err; /* Port number out of range */
+    }
+
+    *p_port_num = port_num;
+    return UCS_OK;
+
+err:
+    return UCS_ERR_NO_DEVICE;
+}
+
+int uct_ib_device_mtu(const char *dev_name, uct_pd_h pd)
```

Comment:
would be better to return `ucs_status_t` and add a parameter `size_t *mtu_p`
because we are always returning error codes as ucs_status_t and not as negative numbers